bin/olly.ml

```diff
@@ -171,30 +190,39 @@ let trace format trace_filename exec_args =
       in
       let init () = () in
       let cleanup () = Trace.close trace_file in
-      olly ~runtime_begin ~runtime_end ~init ~cleanup exec_args
+      olly ~runtime_begin ~runtime_end ~init ~lifecycle ~cleanup exec_args
 
-let latency json output exec_args =
+let gc_stats json output exec_args =
   let current_event = Hashtbl.create 13 in
   let hist =
     H.init ~lowest_discernible_value:10 ~highest_trackable_value:10_000_000_000
       ~significant_figures:3
   in
+  let is_gc_phase phase =
+    match phase with
+    | "major" | "stw_leader" | "interrupt_remote" -> true
+    | _ -> false
+  in
   let runtime_begin ring_id ts phase =
     match Hashtbl.find_opt current_event ring_id with
-    | None -> Hashtbl.add current_event ring_id (phase, Ts.to_int64 ts)
+    | None when
+      (is_gc_phase (Runtime_events.runtime_phase_name phase)) ->
```

Comment:
For performance reasons, better to not convert to string and then do compare. You should directly compare the event name constructor in `is_gc_phase`: https://v2.ocaml.org/releases/5.0/api/Runtime_events.html#TYPEruntime_phase. This would be much faster as there is no need to allocate the string and do the o(n) string comparison. Constructor comparisons are O(1). 
It would be useful to move this comparison outside the hash table test for performance reasons.